sympy/physics/continuum_mechanics/cross_section.py

```diff
@@ -0,0 +1,161 @@
+from sympy.geometry.polygon import Polygon, Point, Circle
+from sympy.core import sympify
+from sympy.printing import sstr
+from sympy.core import Symbol, S
+
+
+class CrossSection(object):
+    def __init__(self, shape, variable, *dimensions):
+        self.shape = shape
+        self.dimensions = dimensions
+        self.variable = variable
+
+        shape = shape.lower()
+        
+        if shape == "circular":
+            self._circular(self.dimensions)
+        elif shape == "triangular":
+            self._triangular(self.dimensions)
+        elif shape == "rectangular":
+            self._rectangular(self.dimensions)
+        elif shape == "tsection":
+            self._tsection(self.dimensions)
+        elif shape == "isection":
+            self._isection(self.dimensions)
+        else:
+            raise ValueError("No such cross-sectional shape is supported")
+
+    def __str__(self):
+        str_1 = 'CrossSection({}, {})'.format(sstr(self._shape), sstr(self._dimensions))
+        return str_1
+
+    __repr__ = __str__
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @shape.setter
+    def shape(self, shape):
+        self._shape = shape
+    
+    @property
+    def dimensions(self):
+        return self._dimensions
+
+    @dimensions.setter
+    def dimensions(self, dim):
```

Comment:
I don't know much about this module, but as far as I know allowing the dimensions to be changed after creating the object may cause wrong results or confusion for the user. May be, you should allow setting the dimensions only once.